wazo_dird/plugins/phonebook_backend/http.py

```diff
@@ -36,3 +50,79 @@ def get(self, source_uuid):
     @required_acl('dird.backends.phonebook.sources.{source_uuid}.update')
     def put(self, source_uuid):
         return super().put(source_uuid)
+
+
+class PhonebookSourceInfo(SourceConfig):
+    phonebook_uuid: str
+
+
+class PhonebookContactList(AuthResource):
+    BACKEND = 'phonebook'
+    _phonebook_service: _PhonebookService
+    _source_service: _SourceService
+
+    def __init__(
+        self, source_service: _SourceService, phonebook_service: _PhonebookService
+    ):
+        self._source_service = source_service
+        self._phonebook_service = phonebook_service
+
+    @required_acl('dird.backends.phonebook.sources.{source_uuid}.contacts.read')
+    def get(self, source_uuid):
+        visible_tenants = get_tenant_uuids(recurse=False)
+        list_params = contact_list_schema.load(request.args)
+        count_params = contact_list_schema.load_count(request.args)
+        source_config = cast(
+            PhonebookSourceInfo,
+            self._source_service.get(self.BACKEND, str(source_uuid), visible_tenants),
+        )
+        logger.debug("Found source (uuid=%s): %s", source_uuid, str(source_config))
+
+        # sources prior to WAZO-3267 might be missing 'phonebook_uuid' and will crash here
```

Comment:
We should add a migration script to remove the invalid sources and remove this comment.

Reply:
Add ticket in comment : https://wazo-dev.atlassian.net/browse/WAZO-3357